fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java

```diff
@@ -2105,6 +2105,15 @@ protected OlapTable selectiveCopyInternal(OlapTable copied, Collection<String> r
     public Partition replacePartition(Partition newPartition) {
         Partition oldPartition = nameToPartition.remove(newPartition.getName());
 
+        // For cloud native table, add partition into recycle Bin after truncate table.
+        // It is no necessary for share nothing mode because file will be deleted throught
+        // tablet report in this case.
+        if (this.isCloudNativeTable()) {
```

Comment:
shall be applicable to shared-data materialized view as well?

Reply:
Updated